### DISCUSSION TITLE
Welcome to Universal_HamRadio_Remote_HTML5 Discussions!

### DISCUSSION THREAD

**Discussion by F4HTB** [F4HTB/Universal_HamRadio_Remote_HTML5#24] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

**Comment by 52amc** [comment]
Greetings All,
First of all i'm not a command line person. Trying to install Universal HamRasio Remote program on the pi
Greetings All,
First of all I’m not a command line person. Trying to install Universal Ham Radio   Remote program on the pi.
I copy and pasted all commands. When it finished the pi said
pi@Raspberry:~/Universal_HamRadio_Remote_HTML5 $ External technical purpose
bash: External: command not found
pi@Raspberry:~/Universal_HamRadio_Remote_HTML5 $ hardware requirements
bash: hardware: command not found
pi@Raspberry:~/Universal_HamRadio_Remote_HTML5 $ History
bash: History: command not found
pi@Raspberry:~/Universal_HamRadio_Remote_HTML5 $ Interface explainnation
bash: Interface: command not found
pi@Raspberry:~/Universal_HamRadio_Remote_HTML5 $ Optional Authentification
bash: Optional: command not found
pi@Raspberry:~/Universal_HamRadio_Remote_HTML5 $ Panadapter
bash: Panadapter: command not found
pi@Raspberry:~/Universal_HamRadio_Remote_HTML5 $ Software installation
bash: Software: command not found
pi@Raspberry:~/Universal_HamRadio_Remote_HTML5 $ ToDo List
bash: ToDo: command not found
pi@Raspberry:~/Universal_HamRadio_Remote_HTML5 $ tunneling compatible for exterior access
bash: tunneling: command not found
pi@Raspberry:~/Universal_HamRadio_Remote_HTML5 $ Clone this wiki locally
bash: Clone: command not found
pi@Raspberry:~/Universal_HamRadio_Remote_HTML5 $ ^C
pi@Raspberry:~/Universal_HamRadio_Remote_HTML5 $ ^C
pi@Raspberry:~/Universal_HamRadio_Remote_HTML5 $

Were do I go from here I’m stuck     Thanks KD4DN Allen

**Comment by bpehrson** [comment]
I made what seemed to be a successful installation following the instruction labeled  "example of complete installation"  on the web. I followed the instruction exactly and everything seemed to go well.
However, when I started the execution, the response was:
sa0bxi@remote:~/Universal_HamRadio_Remote_HTML5 $ PYTHONPATH=/usr/local/lib/python3.7/site-packages:$PYTHONPATH ./UHRR
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/Hamlib.py", line 14, in swig_import_helper
    return importlib.import_module(mname)
  File "/usr/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 670, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 583, in module_from_spec
  File "<frozen importlib._bootstrap_external>", line 1043, in create_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
ImportError: libhamlib.so.4: cannot open shared object file: No such file or directory

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "./UHRR", line 18, in <module>
    import Hamlib
  File "/usr/local/lib/python3.7/site-packages/Hamlib.py", line 17, in <module>
    _Hamlib = swig_import_helper()
  File "/usr/local/lib/python3.7/site-packages/Hamlib.py", line 16, in swig_import_helper
    return importlib.import_module('_Hamlib')
  File "/usr/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
ImportError: libhamlib.so.4: cannot open shared object file: No such file or directory
sa0bxi@remote:~/Universal_HamRadio_Remote_HTML5 $

Any advice?

**Comment by kt4wo** [comment]
Is this project still viable ??  And will this run on a Pi2B??  or do I 
need the Pi3 and up??
Looking forward to trying this out!! 
Thanks de Trip - KT4WO

**Comment by F4HTB** [comment]
Hello, 

**Is this project still viable ??** 
=>If by viable you mean it works like proof of concept, yes.
Lot of people download and try every days but i have only few feedback on real operation.
But remember that this is an open source project and that I don't have many contributors...

**And will this run on a Pi2B?? or do I need the Pi3 and up??**
=>I only test and dev on a pi4 for this project.
It seems that on a pc it works but an rpi3 seems at the limit.

**Looking forward to trying this out!!
Thanks de Trip - KT4WO**

73 F4HTB

**Comment by 52amc** [comment]
Hello I personally have no luck with it. Its  been a year since I tried it.
I tried loading it on a pi4.
Looks like it’s a well written program . I’m by no means a pi person.
But I was able to load and run most of the pi programs for digital work.

**Comment by kt4wo** [comment]
On 5/2/22, Olivier SCHMITT ***@***.***> wrote:
> Hello,
>
> **Is this project still viable ??**
> =>If by viable you mean it works like proof of concept, yes.


Hey Olivier,

That's great!
Right now I am working with Simon - DJ2LS on the FreeDATA
"alpha"  testing. I cannot code but am OK testing stuff, so
when I get some time maybe I can jump in and start
testing/running UHRR.
I have ALL the RPi's so will start with the 3B and work
my way up.
Thanks for your work on the software!!  I am a huge fan
of FOSS software for Amateur Radio.

Trip - KT4WO    EM85ii

**Comment by lbrighten** [comment]
1. I installed the UHRR software according to directions in Example of complete Installation. However, when I issued the Run code section:
cd ~/Universal_HamRadio_Remote_HTML5
PYTHONPATH=/usr/local/lib/python3.7/site-packages:$PYTHONPATH ./UHRR

I received an error stating it couldn't find Hamlib. I noticed that the version of python was updated to python3.9. I changed the PYTHONPATH to reflect that version and the web page successfully started.
2. A command window on the Rpi shows the server activity. I am using an Android smartphone with a Chrome browser and can connect successfully to my Icom IC-7410 transceiver. The on screen controls all seem to work okay but when I press the Tx button, the radio does not respond and the command window repeats "overrun" and if I stop Tx and press it again then it actually puts the rig in transmit mode and the command window shows: "Send to all: getPTT:True. The lower right corner of the screen shows latency. When the overrun occurs the latency is up over 1500ms. Most of the time it is around 50 ms. Is there any adjustments that can be made to stop that? I'm using my local wireless on the phone and am setting right next to the access point.
3. I'd like to use the app to talk to my 2 meter radio. The audio connection would be to just use a USB Audio dongle on the pi, but how do I get PTT? I'd like to use one of the GPIO lines on the pi to control PTT on the radio.

**Comment by somervda** [comment]
F4HTB - This is a fantastic bit of code, I was looking to go down the similar route (A Hamlib based web service interface, but using an Angular frontend ) and so I had started to look at tornado as the interface between Hamlib and a python async/react web service provider. Looks like you have, been there and done a lot of this already :) .

**Comment by Balaganoff** [comment]
RP3 +  Raspbian GNU/Linux 11 (bullseye)
Everything is installed, the server starts, but the error appears 

SERVER port 8888
SERVER auth
SERVER db_users_file UHRR_users.db
AUDIO outputdevice plughw:CARD=Device,DEV=0
AUDIO inputdevice plughw:CARD=Device,DEV=0
HAMLIB rig_model FT817
HAMLIB rig_pathname /dev/ttyUSB0
HAMLIB rig_rate 38400
HAMLIB trxautopower True
HAMLIB data_bits
HAMLIB stop_bits
HAMLIB serial_parity
HAMLIB serial_handshake
HAMLIB dtr_state
HAMLIB rts_state
PANADAPTER center_freq 68330000
PANADAPTER sample_rate 960000
PANADAPTER freq_correction 1
PANADAPTER gain 10
PANADAPTER fft_window hamming
recording...
(<class 'AttributeError'>, AttributeError("module 'Hamlib' has no attribute 'hamlib_port_parm_serial'"), <traceback object at 0x73967c88>)
HTTP server started.



lsusb output
Bus 001 Device 006: ID 0dd8:3b00 Netac Technology Co., Ltd OnlyDisk
Bus 001 Device 005: ID 08bb:2902 Texas Instruments PCM2902 Audio Codec
Bus 001 Device 004: ID 067b:2303 Prolific Technology, Inc. PL2303 Serial Port / Mobile Action MA-8910P
Bus 001 Device 003: ID 0424:ec00 Microchip Technology, Inc. (formerly SMSC) SMSC9512/9514 Fast Ethernet Adapter
Bus 001 Device 002: ID 0424:9514 Microchip Technology, Inc. (formerly SMSC) SMC9514 Hub
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub

This USB cable works under Windows in a bundle FT817 + FT817 Commander

Who can help me please!

**Reply by Balaganoff** [reply]
I'll answer myself. Maybe it will be useful to someone else.
This is my way with RP3 + Pi OS 64-bit (Debian 12 Bookworm) +FT 817. In the end, everything works. Below I provide a step-by-step instruction.

Many thanks to the author for the work done!

```
# 📌 UHRR Setup Guide: Universal Ham Radio Remote

## 1. Install System Dependencies
sudo apt update
sudo apt install -y \
    git python3 python3-pip python3-numpy python3-tornado \
    python3-serial python3-pyaudio rtl-sdr \
    libasound2-dev python3-dev \
    libopus-dev libtool autoconf automake swig \
 

## 2. Remove Conflicting Hamlib Versions
# Clean up any manually compiled Hamlib to avoid conflicts
sudo rm -f /usr/local/bin/rigctl
sudo rm -rf /usr/local/lib/libhamlib.*
sudo rm -rf /usr/local/lib/python3.11/site-packages/Hamlib*
sudo ldconfig

## 3. Install System Hamlib (from apt)
# Use Debian's stable version instead of git or PyPI
sudo apt install -y hamlib-utils python3-hamlib

## 4. Install Python Packages
# Required for audio, SDR, WebSocket, and control
pip install --break-system-packages \
    tornado opuslib pyalsaaudio pyrtlsdr pyserial pam

## 5. Clone the Project
cd ~
git clone https://github.com/F4HTB/Universal_HamRadio_Remote_HTML5.git
cd ~/Universal_HamRadio_Remote_HTML5

## 6. Set Permissions & Latency Timer
# Add user to dialout for serial access, set low latency for FTDI
sudo usermod -aG dialout,plugdev,audio $USER
echo 1 | sudo tee /sys/bus/usb-serial/devices/ttyUSB0/latency_timer

## 7. Reboot
# Required for group changes and latency_timer to take effect
sudo reboot

## 8. Verify Devices
# Check if devices are accessible
ls -l /dev/ttyUSB0                    # Should be owned by dialout
python3 -c "import alsaaudio; print(alsaaudio.pcms())"  # List audio devices
rtl_test -t                           # Test RTL-SDR

## 9. Configure UHRR.conf
# Edit UHRR.conf to match your hardware
# [HAMLIB]
# rig_pathname = /dev/ttyUSB0
# rig_rate = 38400
# rig_model = FT817
#
# [AUDIO]
# inputdevice = plughw:CARD=Device,DEV=0
# outputdevice = plughw:CARD=Device,DEV=0

## 10. Run UHRR
# Start the server
./UHRR

## 11. Troubleshooting
# Use these commands if UHRR fails to start or connect

# Kill old processes
killall python3 pulseaudio || true
sudo rmmod dvb_usb_rtl28xxu || true
pulseaudio --kill && pulseaudio --start

# Test Hamlib import
python3 -c "import Hamlib; print(Hamlib.RIG_MODEL_FT817)"
# Expected output: 1020

# Test rigctl communication
rigctl -m 1020 -r /dev/ttyUSB0 -s 38400 f
# Should return current frequency
```

**Comment by F4HTB** [comment]
Hello Dear,

Thank you very much, your feedback is greatly appreciated. It's through 
participation that the project can evolve.
With your permission, I will use your information to complete a future 
installation wiki page.

73 F4HTB.

Le 03/09/2025 à 09:34, Balaganoff a écrit :

**Reply by Balaganoff** [reply]
Hi, Olivier!
Of course, I don't mind! Use my information as you want and where you want! ;)
Thanks again for the interesting project! I wish it further development!
73!
